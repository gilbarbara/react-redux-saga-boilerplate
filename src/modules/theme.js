export const headerHeight = 70;

export const appColor = '#00b4d5';

export const easing = 'cubic-bezier(0.35, 0.01, 0.77, 0.34);';

export default {
  breakpoints: [400, 768, 1024, 1280, 1920],
  palette: {
    primary: appColor,
  },
  button: {
    borderRadius: {
      xs: 4,
      lg: 28,
      xl: 32,
    },
    padding: {
      lg: [12, 28],
      xl: [14, 32],
    },
  },
};
