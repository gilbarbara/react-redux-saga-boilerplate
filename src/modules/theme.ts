import { mergeTheme } from '@gilbarbara/components';

export const headerHeight = 74;

export const appColor = '#00b4d5';

export const easing = 'cubic-bezier(0.35, 0.01, 0.77, 0.34);';

const theme = mergeTheme({
  button: {
    xs: {
      borderRadius: '4px',
    },
    lg: {
      borderRadius: '32px',
      padding: ['14px', '32px'],
    },
  },
});

export const variants = theme.colors;

export default theme;
