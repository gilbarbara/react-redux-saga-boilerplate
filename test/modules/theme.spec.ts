import theme from '~/modules/theme';

describe('modules/theme', () => {
  describe('theme', () => {
    it('should return the theme', () => {
      expect(theme).toMatchSnapshot();
    });
  });
});
