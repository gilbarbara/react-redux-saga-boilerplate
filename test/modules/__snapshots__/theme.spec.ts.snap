// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`modules/theme > theme > should return the theme 1`] = `
{
  "avatar": {
    "jumbo": {
      "fontSize": "52px",
      "size": "128px",
    },
    "lg": {
      "fontSize": "24px",
      "size": "64px",
    },
    "md": {
      "fontSize": "18px",
      "size": "48px",
    },
    "sm": {
      "fontSize": "14px",
      "size": "32px",
    },
    "xl": {
      "fontSize": "40px",
      "size": "96px",
    },
    "xs": {
      "fontSize": "12px",
      "size": "24px",
    },
    "xxs": {
      "fontSize": "8px",
      "size": "16px",
    },
  },
  "black": "#000",
  "breakpoints": {
    "lg": "1024px",
    "md": "768px",
    "sm": "400px",
    "xl": "1280px",
    "xs": "375px",
    "xxl": "1440px",
    "xxs": "320px",
    "xxxl": "1680px",
  },
  "button": {
    "lg": {
      "borderRadius": "32px",
      "fontSize": "18px",
      "fontWeight": 700,
      "height": "48px",
      "lineHeight": 1.4,
      "padding": [
        "14px",
        "32px",
      ],
    },
    "md": {
      "borderRadius": "8px",
      "fontSize": "16px",
      "fontWeight": 700,
      "height": "40px",
      "lineHeight": 1.3,
      "padding": [
        "8px",
        "16px",
      ],
    },
    "sm": {
      "borderRadius": "8px",
      "fontSize": "14px",
      "fontWeight": 700,
      "height": "32px",
      "lineHeight": 1.2,
      "padding": [
        "6px",
        "12px",
      ],
    },
    "xs": {
      "borderRadius": "4px",
      "fontSize": "12px",
      "fontWeight": 700,
      "height": "24px",
      "lineHeight": 1.2,
      "padding": [
        "4px",
        "8px",
      ],
    },
  },
  "colors": {
    "blue": "#66a5ff",
    "cyan": "#0bc5ea",
    "gray": "#999999",
    "green": "#58d063",
    "indigo": "#8a8fff",
    "orange": "#ff995d",
    "pink": "#ee63ab",
    "primary": "#3030e8",
    "purple": "#ad7bff",
    "red": "#ff5e5e",
    "secondary": "#2a5d65",
    "teal": "#38b2ac",
    "yellow": "#ffe166",
  },
  "darkColor": "#101010",
  "dropShadow": {
    "high": "drop-shadow(2px 4px 18px rgba(148, 148, 148, 0.8))",
    "low": "drop-shadow(2px 4px 8px rgba(148, 148, 148, 0.4))",
    "mid": "drop-shadow(2px 4px 12px rgba(148, 148, 148, 0.6))",
  },
  "easing": "cubic-bezier(0.35, 0.01, 0.77, 0.34);",
  "fontFamily": "inherit",
  "fontMonospace": "Courier, monospace",
  "fontWeights": {
    "bold": 700,
    "normal": 400,
  },
  "grayScale": {
    "10": "#fcfcfc",
    "100": "#e6e6e6",
    "150": "#d9d9d9",
    "20": "#fafafa",
    "200": "#cccccc",
    "30": "#f7f7f7",
    "300": "#b3b3b3",
    "40": "#f5f5f5",
    "400": "#999999",
    "50": "#f2f2f2",
    "500": "#808080",
    "600": "#666666",
    "700": "#4d4d4d",
    "750": "#404040",
    "800": "#333333",
    "850": "#262626",
    "900": "#1a1a1a",
    "950": "#0d0d0d",
  },
  "inputHeight": {
    "large": "48px",
    "normal": "40px",
  },
  "lightColor": "#f5f5f5",
  "opacity": {
    "intense": 0.64,
    "light": 0.16,
    "medium": 0.32,
    "semiOpaque": 0.72,
    "semiTransparent": 0.08,
  },
  "radius": {
    "lg": "24px",
    "md": "16px",
    "round": "50%",
    "sm": "12px",
    "xl": "32px",
    "xs": "8px",
    "xxs": "4px",
  },
  "shadow": {
    "high": "2px 4px 16px 8px rgba(148, 148, 148, 0.32)",
    "low": "2px 4px 16px 2px rgba(148, 148, 148, 0.16)",
    "mid": "2px 4px 16px 4px rgba(148, 148, 148, 0.24)",
  },
  "spacing": {
    "jumbo": "96px",
    "lg": "24px",
    "md": "16px",
    "sm": "12px",
    "xl": "32px",
    "xs": "8px",
    "xxl": "48px",
    "xxs": "4px",
    "xxxl": "64px",
  },
  "typography": {
    "h1": {
      "fontSize": "32px",
      "lineHeight": 1.2,
    },
    "h2": {
      "fontSize": "28px",
      "lineHeight": 1.2,
    },
    "h3": {
      "fontSize": "24px",
      "lineHeight": 1.2,
    },
    "h4": {
      "fontSize": "20px",
      "lineHeight": 1.2,
    },
    "h5": {
      "fontSize": "18px",
      "lineHeight": 1.2,
    },
    "h6": {
      "fontSize": "16px",
      "lineHeight": 1.2,
    },
    "jumbo": {
      "fontSize": "48px",
      "lineHeight": 1.2,
    },
    "jumboLarge": {
      "fontSize": "64px",
      "lineHeight": 1.2,
    },
    "lg": {
      "fontSize": "18px",
      "lineHeight": 1.2,
    },
    "md": {
      "fontSize": "16px",
      "lineHeight": 1.2,
    },
    "sm": {
      "fontSize": "14px",
      "lineHeight": 1.2,
    },
    "xs": {
      "fontSize": "12px",
      "lineHeight": 1.2,
    },
  },
  "variants": {
    "blue": {
      "100": "#cce1ff",
      "200": "#99c3ff",
      "300": "#66a5ff",
      "400": "#3788fb",
      "50": "#e6f0ff",
      "500": "#056afa",
      "600": "#0054cc",
      "700": "#003f99",
      "800": "#002a66",
      "900": "#001533",
    },
    "cyan": {
      "100": "#ccf6ff",
      "200": "#9aeefe",
      "300": "#6ae3fb",
      "400": "#3cd7f6",
      "50": "#e6fbff",
      "500": "#0bcdf4",
      "600": "#05a7c7",
      "700": "#017f98",
      "800": "#005566",
      "900": "#002a33",
    },
    "gray": {
      "100": "#e6e6e6",
      "200": "#cccccc",
      "300": "#b3b3b3",
      "400": "#999999",
      "50": "#f2f2f2",
      "500": "#808080",
      "600": "#666666",
      "700": "#4d4d4d",
      "800": "#333333",
      "900": "#1a1a1a",
    },
    "green": {
      "100": "#d3f8d7",
      "200": "#abedb2",
      "300": "#85e08c",
      "400": "#60d26a",
      "50": "#e7fde8",
      "500": "#38c745",
      "600": "#29a334",
      "700": "#1c7d24",
      "800": "#0e5816",
      "900": "#033008",
    },
    "indigo": {
      "100": "#ccceff",
      "200": "#999dff",
      "300": "#6970fc",
      "400": "#3b43f7",
      "50": "#e6e7ff",
      "500": "#0a14f5",
      "600": "#040dc8",
      "700": "#000799",
      "800": "#000466",
      "900": "#000233",
    },
    "orange": {
      "100": "#ffdfcc",
      "200": "#ffbf99",
      "300": "#ff9f66",
      "400": "#fb8037",
      "50": "#ffefe6",
      "500": "#fa5f05",
      "600": "#cc4c00",
      "700": "#993900",
      "800": "#662600",
      "900": "#331300",
    },
    "pink": {
      "100": "#fdcee6",
      "200": "#f7a1cd",
      "300": "#f075b5",
      "400": "#e74b9c",
      "50": "#ffe6f3",
      "500": "#e11e83",
      "600": "#b81469",
      "700": "#8d0c4f",
      "800": "#620434",
      "900": "#33001a",
    },
    "primary": {
      "100": "#cdcdfe",
      "200": "#9f9ff9",
      "300": "#7272f3",
      "400": "#4747eb",
      "50": "#e5e5ff",
      "500": "#1a1ae6",
      "600": "#1010bc",
      "700": "#090990",
      "800": "#020264",
      "900": "#000033",
    },
    "purple": {
      "100": "#dfccff",
      "200": "#c099ff",
      "300": "#a066ff",
      "400": "#8137fb",
      "50": "#efe6ff",
      "500": "#6205fa",
      "600": "#4d00cc",
      "700": "#3a0099",
      "800": "#270066",
      "900": "#130033",
    },
    "red": {
      "100": "#ffcccc",
      "200": "#ff9999",
      "300": "#ff6666",
      "400": "#fb3737",
      "50": "#ffe6e6",
      "500": "#fa0505",
      "600": "#cc0000",
      "700": "#990000",
      "800": "#660000",
      "900": "#330000",
    },
    "secondary": {
      "100": "#d9eef2",
      "200": "#b7dbe1",
      "300": "#96c8cf",
      "400": "#77b2bb",
      "50": "#eaf8fb",
      "500": "#559faa",
      "600": "#40828c",
      "700": "#2d646c",
      "800": "#1a454c",
      "900": "#09252a",
    },
    "teal": {
      "100": "#d4f7f4",
      "200": "#adebe8",
      "300": "#88ddd9",
      "400": "#64cec9",
      "50": "#e8fdfb",
      "500": "#3dc2bb",
      "600": "#2d9f9a",
      "700": "#1f7a76",
      "800": "#105652",
      "900": "#042f2c",
    },
    "yellow": {
      "100": "#fff5cc",
      "200": "#ffeb99",
      "300": "#ffe166",
      "400": "#fbd537",
      "50": "#fffae6",
      "500": "#faca05",
      "600": "#cca400",
      "700": "#997b00",
      "800": "#665200",
      "900": "#332900",
    },
  },
  "white": "#fff",
}
`;
